packages/opds-web-client/src/actions.ts

```diff
@@ -83,31 +83,47 @@ export default class ActionCreator {
   }
 
   fetchBlob(type: string, url: string) {
-    return (dispatch): Promise<Blob> => {
+    return async (dispatch): Promise<Blob> => {
```

Comment:
Do we want to throw errors rather than rejecting the error to display some kind of error to the user?

Reply:
I think this does throw an error does it not? Not totally sure I follow you

Reply:
Before, the `catch` would catch any error thrown and return a Promise `reject`. But now this is not resolving or rejecting any values, it's just throwing an error. So just concerned that if that's not caught, the app will crash since an error is thrown, whereas if you catch the error and reject a value, the app can continue working and a friendly error message can be rendered. I'm also not sure how an async promise works without a resolve/reject so maybe it's a lack of knowledge for that.

Reply:
Yea I'm not totally sure either actually now that I think of it. I believe that it should be the same. Previously during failure:

1. Caught in `fetchBlob`, sent `failure` action
2. Rejected the promise. This had to be caught in a parent catch block, or else it results in an unhandled promise rejection error.

Now:

1. Dispatch the `failure` type
2. Rethrow the error. This has to similarly be caught by a parent catch block or else it causes an error.

So I believe it's the same... see here for more info: https://javascript.info/promise-error-handling#unhandled-rejections